kernel/src/expressions/scalars.rs

```diff
@@ -39,6 +33,42 @@ impl ArrayData {
     }
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub struct MapData {
+    data_type: MapType,
+    pairs: Vec<(Scalar, Scalar)>,
+}
+
+impl MapData {
+    pub fn try_new(
+        data_type: MapType,
+        values: impl IntoIterator<Item = (impl Into<Scalar>, impl Into<Scalar>)>,
+    ) -> DeltaResult<Self> {
+        // TODO: check that inner types are primitive?
```

Comment:
Shouldn't we validate (for now) that they are strings, since that's all we support?

Reply:
hm. the `Scalar::to_array` in arrow only supports string->string maps, but I think generally we could still allow for `Scalar::Map` to describe non-string->string maps? But yea I think here we should either (1) check that inner types are strings or (2) check they are primitive? (I think we have that assumption elsewhere so probably should start with such a constraint and relax it later if we can?)

Reply:
did primitive for now (and will constrain pending discussion in https://github.com/delta-io/delta-kernel-rs/pull/881#discussion_r2060638060)